src/compiler/types.ts

```diff
@@ -8219,10 +8219,10 @@ export interface NodeFactory {
     updateJSDocReadonlyTag(node: JSDocReadonlyTag, tagName: Identifier | undefined, comment: string | NodeArray<JSDocComment> | undefined): JSDocReadonlyTag;
     createJSDocUnknownTag(tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocUnknownTag;
     updateJSDocUnknownTag(node: JSDocUnknownTag, tagName: Identifier, comment: string | NodeArray<JSDocComment> | undefined): JSDocUnknownTag;
-    createJSDocDeprecatedTag(tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocDeprecatedTag;
-    updateJSDocDeprecatedTag(node: JSDocDeprecatedTag, tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocDeprecatedTag;
-    createJSDocOverrideTag(tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocOverrideTag;
-    updateJSDocOverrideTag(node: JSDocOverrideTag, tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocOverrideTag;
+    createJSDocDeprecatedTag(tagName: Identifier | undefined, comment?: string | NodeArray<JSDocComment>): JSDocDeprecatedTag;

```

Comment:
Were these changed to match the functions in nodeFactory? I'd rather fix node factory as these functions shouldn't allow `undefined` for `tagName` because the type it creates doesn't allow `undefined`.

Reply:
It's due to `getJSDocSimpleTagCreateFunction`, yeah.

Reply:
Revisiting this; it appears as though effectively all of the JSDoc tags support an undefined tagName, then interally they treat that as "give me the default", either via explicitly doing something like `createIdentifier("param")`, or just calling `getDefaultTagName(node)`.

I don't have a major preference, but I tried making these required and found that we have quite a few cases where we pass in undefined (in services and deprecation) to say "I don't care, give me the most common / defualt".